### PR TITLE
Add ability to define env vars from k8s secrets

### PR DESCRIPTION
**Pull request by andytson-inviqa**
Add ability to define env vars from k8s secrets …

* with sealed-secrets used by default for encryption
* environment variables in templates simplified by merging dicts before iterating
* support for custom CRD schemas necessary to pass tests
* local secrets (plaintext still) excluded from pipeline to avoid confusion over encryption

### CODE REVIEW

File: src/_base/harness/attributes/docker-base.yml [hunk 1]
```diff
@@ -89,6 +92,8 @@ attributes:
         nginx:
           environment:
             FPM_HOST: localhost
+          ingress:
+            annotations: {}
```
Comment (kierenevans) [inline src/_base/harness/attributes/docker-base.yml:96]: Does this still get picked up by `.Values.docker.services.nginx.ingress.annotations` in `src/_base/helm/app/templates/application/webapp/ingress.yaml` and `src/spryker/helm/app/templates/application/webapp/ingress.yaml`?

Comment (andytson-inviqa) [reply src/_base/harness/attributes/docker-base.yml:96]: yes, it's filtered out of workspace attribute merge of services.*, but pipeline.services.* is left unaltered

File: src/spryker/helm/app/templates/application/jenkins-runner/deployment.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
-{{ if .Values.service.jenkins_runner }}
+{{- if .Values.service.jenkins_runner -}}
+{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-base") -}}
```
Comment (kierenevans) [inline src/spryker/helm/app/templates/application/jenkins-runner/deployment.yaml]:
```suggestion
{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "jenkins-runner") (index .Values.docker.services "php-base") -}}
```

File: src/akeneo/helm/app/templates/application/job-queue-consumer/deployment.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{- $service := merge (index .Values.docker.services "console") (index .Values.docker.services "job-queue-consumer") (dict "environment" .Values.environment) -}}
```
Comment (kierenevans) [inline src/akeneo/helm/app/templates/application/job-queue-consumer/deployment.yaml]:
```suggestion
{{- $service := merge (index .Values.docker.services "job-queue-consumer") (index .Values.docker.services "php-base") (dict "environment" .Values.environment) -}}
```

**Review by kierenevans — CHANGES_REQUESTED** [review]
Looking good, just two potential issues

File: src/_base/helm/app/templates/application/webapp/deployment.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+{{- $service_php_fpm := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-fpm") -}}
```
Comment (kierenevans) [inline src/_base/helm/app/templates/application/webapp/deployment.yaml]:
```suggestion
{{- $service_php_fpm := merge (dict "environment" .Values.environment) (index .Values.docker.services "php-fpm") (index .Values.docker.services "php-base") -}}
```

File: src/_base/helm/app/templates/application/cron/deployment.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
-{{ if .Values.service.cron }}
-{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-base") -}}
+{{- if .Values.service.cron -}}
+{{- $service := mergeOverwrite (dict) (index .Values.docker.services "php-base") (index .Values.docker.services "console") (dict "environment" .Values.environment) -}}
```
Comment (kierenevans) [inline src/_base/helm/app/templates/application/cron/deployment.yaml]:
```suggestion
{{- $service := mergeOverwrite (dict) (index .Values.docker.services "php-base") (index .Values.docker.services "cron") (dict "environment" .Values.environment) -}}
```

**Review by kierenevans — CHANGES_REQUESTED** [review]
Last one :)

**Review by kierenevans — APPROVED** [review]

**Pull request merged**